src/iot/commands/createCert.ts

```diff
@@ -0,0 +1,107 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import * as localizedText from '../../shared/localizedText'
+import { getLogger } from '../../shared/logger'
+import { Commands } from '../../shared/vscode/commands'
+import { Window } from '../../shared/vscode/window'
+import { localize } from '../../shared/utilities/vsCodeUtils'
+import { showViewLogsMessage, showConfirmationMessage } from '../../shared/utilities/messages'
+import { IotCertsFolderNode } from '../explorer/iotCertFolderNode'
+import { fileExists } from '../../shared/filesystemUtilities'
+
+/**
+ * Wizard to create a certificate key pair and save them to the filesystem.
+ */
+export async function createCertificateCommand(
+    node: IotCertsFolderNode,
+    window = Window.vscode(),
+    commands = Commands.vscode()
+): Promise<void> {
+    getLogger().debug('CreateCertificate called for %O', node)
+
+    const isConfirmed = await showConfirmationMessage(
+        {
+            prompt: localize('AWS.iot.createCert.prompt', 'Create a new X.509 certificate and RSA key pair?'),
+            confirm: localize('AWS.iot.createCert.confirm', 'Confirm'),
+            cancel: localizedText.cancel,
+        },
+        window
+    )
+    if (!isConfirmed) {
+        getLogger().info('CreateCertificate canceled')
+        return
+    }
+
+    const folderLocation = await promptForFolderLocation(window)
+    if (!folderLocation) {
+        getLogger().info('CreateCertificate canceled: No folder selected')
+        return
+    }
+    const certPath = `${folderLocation.fsPath}-certificate.pem.crt`
+    const privateKeyPath = `${folderLocation.fsPath}-private.pem.key`
+    const publicKeyPath = `${folderLocation.fsPath}-public.pem.key`
```

Comment:
Yes, the hyphens were intentional. When downloaded from the console, IoT saves the files as `{64 character certificate ID}-{certificate|private|public}.pem.{crt|key}`

I considered using the "open" dialog instead, filter for folders and just name them the same way as the AWS console, but decided to use the "save" dialog so that the name of the file could be customized. What do you think is the best option?